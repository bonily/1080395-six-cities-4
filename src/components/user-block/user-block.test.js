import React from "react";
import renderer from "react-test-renderer";
import UserBlock from "../user-block/user-block.jsx";

const name = `Ivan`;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

describe(`UserBlockSnapTest`, () => {
  it(`UserBlock with AuthorizationStatus positiv flag should render user block with the name`, () => {
    const tree = renderer
      .create(<UserBlock
        authorizationStatus = {AuthorizationStatus.AUTH}
        name = {name}
        onUserBlockClick = {() => {}}
      />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`UserBlock with AuthorizationStatus negative flag should render header block with Sign In`, () => {
    const tree = renderer
      .create(<UserBlock
        authorizationStatus = {AuthorizationStatus.NO_AUTH}
        onUserBlockClick = {() => {}}
      />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
