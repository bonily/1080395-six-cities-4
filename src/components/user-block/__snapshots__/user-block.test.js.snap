// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlockSnapTest UserBlock with isLoginComplete negative flag should render header block with Sign In 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="#"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__login"
        >
          Sign in
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`UserBlockSnapTest UserBlock with isLoginComplete positiv flag should render user block with the name 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="#"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__login"
        >
          Ivan
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
